### ISSUE TITLE
NVidia drivers are not found with each new update of steam

### ISSUE THREAD

**Issue by soumyaDghosh** [canonical/steam-snap#347]
I have Nvidia RTX3050 with intel i7 11th gen. In the steam snap with all options maxed out, I get stutters and some frame drops in NFS, but tested with steam deb version. It's not the same there, it works smoothly. The game probably doesn't support mangohud, so, I can't actually calculate/show the fps. I do have a 144hz display, and it's the game play that doesn't feel as it's in the deb version. Any way to give more info/debug this, I'd surely give the details as asked.

**Comment by ashuntu** [comment]
Could you provide the information given from running [`snap run steam.report`](https://github.com/canonical/steam-snap/wiki/Troubleshooting#submitting-a-steam-report)?

Also, do you have `nvidia-smi` available on your system? If so, does NFS appear in `nvidia-smi` after running it? (You can use `nvidia-smi -l` to continuously update)

**Comment by soumyaDghosh** [comment]
Apparently I cannot even run the game anymore from steam snap. Getting this.

```shell

(steam:59657): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
/bin/sh\0-c\0/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=3727512741 -- /home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/steamapps/common/Proton - Experimental'/proton waitforexitandrun  "/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/steamapps/compatdata/4144120982/pfx/drive_c/Mr DJ/Need For Speed Most Wanted Black Edition/speed.exe"\0
chdir "/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/steamapps/compatdata/4144120982/pfx/drive_c/Mr DJ/Need For Speed Most Wanted Black Edition/"
ERROR: ld.so: object '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/soumyaxubuntu/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.

(steam:59657): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libGLX_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libGLX_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libGLX_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libGLX_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libGLX_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.129.03" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu:/snap/steam/171/usr/lib/i386-linux-gnu/pulseaudio:/snap/steam/171/usr/lib/x86_64-linux-gnu/alsa-lib:/snap/steam/171/usr/lib/x86_64-linux-gnu/pulseaudio:/snap/steam/171/graphics/usr/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib/x86_64-linux-gnu:/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau", ld.so.cache, DT_RUNPATH or fallback /lib:/usr/lib
pressure-vessel-wrap[60139]: W: "snap/steam/171/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[60139]: W: "snap/steam/171/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[60139]: W: "var/lib/snapd/hostfs/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[60139]: W: "snap/steam/171/graphics/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[60139]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
pressure-vessel-wrap[60139]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
Uploaded AppInterfaceStats to Steam
```

This is the `steam.report`

```shell
os_release:
    name:               "Ubuntu"
    version:            "23.10 (Mantic Minotaur)"
snap_info:
    steam_revision:     171
    snapd_revision:     20671
lspci:
glxinfo:
    gpu:                Mesa Intel(R) UHD Graphics (TGL GT1)
    gpu_version:        4.6 (Core Profile) Mesa 23.2.1 refresh - kisak-mesa PPA
lscpu:
    model_name:         11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu-waylandos_release:
    name:               "Ubuntu"
    version:            "23.10 (Mantic Minotaur)"
snap_info:
    steam_revision:     171
    snapd_revision:     20671
lspci:
glxinfo:
    gpu:                Mesa Intel(R) UHD Graphics (TGL GT1)
    gpu_version:        4.6 (Core Profile) Mesa 23.2.1 refresh - kisak-mesa PPA
lscpu:
    model_name:         11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu-wayland

```

Can run the game from the deb.

**Comment by ashuntu** [comment]
Does running `sudo /usr/lib/snapd/snap-discard-ns steam` (with Steam closed), and then running the Steam and the game change things at all?

If not, what about restarting your computer?

Failing all that, what about uninstalling and reinstalling the Steam snap?

We've had scattered reports before of GPU driver updates breaking the Snap until doing the above, so these are just workarounds.

**Comment by soumyaDghosh** [comment]
I had to reinstall steam. But, this time, I installed it from the edge version and the game seemed to be better than before. Closing this issue for now.

**Comment by soumyaDghosh** [comment]
@ashuntu with each new update of steam, the nvidia gpu is breaking. I think this is an issue

**Comment by soumyaDghosh** [comment]
I am still facing this issue. Requesting @ashuntu to reopen this

**Comment by gazhay** [comment]
Every driver update causes steam to fail to detect vulkan.

Reinstalling the snap is not a solution as if you don't purge it essentially gzips every game and takes years.
If you do purge then you have to reinstall every game again. 

At the moment even a reinstall has not fixed my vulkan, i've spent more time trying to get this running that I played games in the last 3 weeks.

**Comment by ashuntu** [comment]
This is especially important at the moment due to changes coming in the snap and snapd, what channels/versions of the snap are you running?

`snap info steam`
`snap info snapd`

**Comment by ashuntu** [comment]
Since there hasn't been an update in a while, and this may be resolved by snapd 2.64 anyway, I'll close this issue for now. Feel free to reopen if it's still a problem.

To test snapd, you can switch to edge by running `snap refresh snapd --edge` (then a restart is advised). Make sure to revert to stable after testing with `snap refresh snapd --stable`.
